Sources/AsyncHTTPClient/HTTPHandler.swift

```diff
@@ -497,13 +497,15 @@ extension HTTPClient {
         var cancelled: Bool
         let lock: Lock
         let id = UUID()
+        let poolingTimeout: TimeAmount?
 
-        init(eventLoop: EventLoop) {
+        init(eventLoop: EventLoop, poolingTimeout: TimeAmount? = nil) {
```

Comment:
Let me know if you think this isn't correct. Since the `Task` has the code that removes / adds back the handlers on release, it needs to know the timeout, so the only other solution I see to avoid passing it to the `Task` would be to refactor the code to move this responsibility outside of `Task`

Reply:
I think currently there's a race risk here. What if

T1: decides to pick a certain `Connection` from the pool
T2 (EventLoop associated with the connection's `Channel`): timeout fires, connection closes
T1: tries to send a request on the connection's `Channel`.

However, I _think_ we can fix this. If we do the following logic:

- select a `Connection` to use
- hop over to to the connection's `Channel.eventLoop` and in _there_ check that the channel `isActive` and remove the handlers
- if the channel was active, continue, if not, go back and select another connection

Reply:
You are right, I will attempt this approach, thanks

Reply:
@weissi Tell me what you think of https://github.com/swift-server/async-http-client/pull/170/commits/2044699c3b2d0e6713b687eee6ab6f4da71d5a88 🙂